# Задачи для лекций по темам числа, логические типы данных, переменные и логические конструкции

## Задача 1.1

Обозначьте порядок вычисления выражения по операциям:
11 * 2 ** 2 – 13 / 4 + 7.

Какой будет ответ?
Какое целое число получается?


In [3]:
a=2**2
a=11*a
b=13/4
c=a-b
c=c+7
int(c)

47

## Задача 1.2

Используя стандартные арифметические операции получите наибольшее число в результате комбинации цифр 4, 6 3

In [28]:
a=np.zeros(3)
a[0]=4
a[1]=6
a[2]=3
a1=a[0]**a[1]**a[2]
a2=a[1]**a[0]**a[2]
list=[a1,a2]
np.max(list)

            


1.109067877648326e+130

## Задача 1.3

Решить пример:  

37052.28 : 76 − 49.03

In [4]:
37052.28/76-49.03

438.5

## Задача 1.4

запишите число Эйлера (2.718281828459045) в экспоненциальной форме 

In [6]:
import numpy as np
np.exp(1)

2.718281828459045

## Задача 1.5

Запишите число 3141592e-6 в виде десятичной дроби и проверьте равны ли они

In [7]:
print(3141592*(10**-6))
print('pi: ', np.pi)

3.1415919999999997
pi:  3.141592653589793


## Задача 1.6

Решите пример: 6 ** 34 - int(float(6 ** 34)). Какой получили ответ и какой ответ должен был получиться?

In [29]:
print(6**34-int(float(6**34)))
print('а должно было быть 0')

17179869184
а должно было быть 0


## Задача 1.7

Решите пример: 0.1 + 0.2. Проверьте, равен ли результат 0.3

In [31]:
print(0.1+0.2)
print('должно было быть 0.3')

0.30000000000000004
должно было быть 0.3


## Задача 1.8

Решить логарифмическое уравнение: 
log5(5 - x) = 2 log5(3)

In [50]:
print(-3**2+5)
import numpy as np

import math as m
import sympy as sp
from sympy.solvers import solve
from sympy import Symbol
x = Symbol('x')
x=solve(sp.log(5-x,5)-2*sp.log(3,5), x)

print(x)

-4
[-4]


## Задача 1.9

Найти объем конуса с pi = 3,14 и pi из библиотеки math, и сравнить результаты. R = 10, h = 8

In [59]:
import math as m
R = 10
h = 8
s1 = ((m.pi*R**2*h)/3)
s2 = ((3.14*R**2*h)/3)
s1 == s2

False

## Задача 1.10  
Дано 2 скрипта. Запустите оба, в каком возникает ошибка? Объясните почему

Скрипт 1.  
a = None  
if len(a) > 0 and a is not None:  
<p style="margin-left: 20px">print('Это не None')</p>

Скрипт 2.  
a = None  
if a is not None and len(a) > 0:  
<p style="margin-left: 20px">print('Это не None')</p>

In [53]:
a = None  
if len(a) > 0 and a is not None:  
    print('Это не None')

TypeError: object of type 'NoneType' has no len()

In [55]:
a = None
if a is not None and len(a) > 0:

    print('Это не None')

он идет последовательно во втором случае до проверки длины не дошло, тк а не none и после and второе условие можно уже не проверять, а в первом случае сначала не смогли проверить длину и вышла ошибка

## Задача 1.11

Написать программу, проверяющую выполнение условия для введённого пользователем числа.  
Условия:  
(-∞, -10] U [-5, 5] U (2, +∞)

In [63]:
a=float(input())
if a<=-10 or (-5<=a<=5) or a>=2:
    print('Число удовлетворяет условию')
else:
    print('Число НЕ удовлетворяет условию')

-6
Число НЕ удовлетворяет условию


## Задача 1.12 \*

Напишите программу, которая будет принимать аргументы и реализовывать алгоритм угадывания даты рождения и возраста: http://www.scanword.net/focus/124.htm

In [67]:
day=int(input('day'))
day=day*2
day=day+5
day=day*50
month=int(input('month'))
day=day+month
print(day)

day4
month5
655


In [68]:

num = int(input())
num = num - 250
print('месяц: ', num%100)
print('день: ', num//100)

655
месяц:  5
день:  4


## Задача 1.13 \*

Напишите программу для определения какой пиццы больше: двух средних или одной большей или они равны (пользователь сам задаёт диаметр каждого типа пиццы)


In [69]:
d1 = int(input('диаметр средней пиццы: '))
d2 = int(input('диаметр большой пиццы: '))
s1=m.pi*(d1/2)**2*2
s2=m.pi*(d2/2)**2*1
print('площадь двух средних: ',s1)
print('площадь одной большой: ',s2)
if s1>s2:
    print('две средние больше')
else:
    print('большая больше')

диаметр средней пиццы: 35
диаметр большой пиццы: 40
площадь двух средних:  1924.2255003237483
площадь одной большой:  1256.6370614359173
две средние больше
